modules/renter/dotsia/format.go

```diff
@@ -0,0 +1,392 @@
+/*
+Package dotsia defines the .sia format.
+
+Specification
+
+A .sia file is a gzipped tar archive containing one or more Files. Each
+File is a JSON representation of the File type defined in this package. For
+each file object header in the tar archive, only the Size field is
+populated. A File only contains metadata, not the actual file as uploaded
+to the Sia network; thus, metadata pertaining to the uploaded file, such as
+its path and mode bits, are specified inside the File, not the tar header.
+
+The first entry in the tar archive is a special metadata file. It is a JSON
+object corresponding to the Metadata type exported by this package. This
+object contains a version string, which indicates the version of the .sia
+format used. At this time, the .sia format has no promise of backwards or
+forwards compatibility, except that the version field will never be removed
+from the metadata object.
+
+The JSON encoding tries to be flexible in allowing arbitrary encryption and
+encoding schemes. As such, the "masterKey" and "erasureCode" fields are
+encoded as generic JSON objects (in Go, a map[string]interface{}). However,
+these objects must always contain a "name" field that identifies the scheme
+used. They may not be null.
+
+Integer fields must not contain negative values.
+
+The "permissions" field is encoded as a decimal number (not octal, or a
+symbolic string) and must not exceed 511 (0777 in octal).
+
+The "path" string must be an absolute Unix-style path; that is, it must begin
+with a leading slash and use '/' as its separator. Additionally, paths must
+not end in a slash, and must not contain any occurences of the current
+directory (.) or parent directory (..) elements.
```

Comment:
This could be changed to a generic "name," such that path-style names would be optional. But I think it's common enough to enforce in the standard. I believe the .zip format also does this.